include/zephyr/devicetree.h

```diff
@@ -435,6 +435,44 @@
  */
 #define DT_CHILD(node_id, child) UTIL_CAT(node_id, DT_S_PREFIX(child))
 
+/**
+ * @brief Get a child node by index
+ *
+ * This macro retrieves a child node of a specified node by its
+ * index. The index corresponds to the order in which the child nodes
+ * are defined in the Devicetree source.
+ *
+ * Example devicetree fragment:
+ *
+ * @code{.dts}
+ * / {
+ *     parent_node {
+ *         first_child {
+ *             reg = <0>;
+ *         };
+ *         second_child {
+ *             reg = <1>;
+ *         };
+ *     };
+ * };
```

Comment:
and what if they are defined in different dtsi files? this is really just confusing